File: 2026-02/src/day_19/boj_2022/Main.cpp
#include <bits/stdc++.h>
using namespace std;

double lower_bound_param(double x, double y, double c) {
    double left = 0;
    double right = min(x, y);

    while (right - left > 0.001) {
        double mid = (left + right) / 2;

        double h1 = sqrt(x * x - mid * mid);
        double h2 = sqrt(y * y - mid * mid);

        if (c < h1 * h2 / (h1 + h2)) {
            left = mid;
        } else {
            right = mid;
        }
    }

    return right;
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    double x, y, c;
    cin >> x >> y >> c;
    cout << lower_bound_param(x, y, c);
}

File: 2026-02/src/day_19/boj_2022/Main.java
package day_19.boj_2022;

import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        double x = Double.parseDouble(st.nextToken());
        double y = Double.parseDouble(st.nextToken());
        double c = Double.parseDouble(st.nextToken());

        System.out.println(lowerBound(x, y, c));
    }

    static double lowerBound(double x, double y, double c) {
        double left = 0;
        double right = Math.min(x, y);

        while (right - left > 1E-3) {
            double mid = (left + right) / 2;

            double h1 = Math.sqrt(x * x - mid * mid);
            double h2 = Math.sqrt(y * y - mid * mid);

            if (c < h1 * h2 / (h1 + h2)) {
                left = mid;
            } else {
                right = mid;
            }
        }

        return right;
    }
}
